test_data/pil/expand_fixed.pil

```diff
@@ -0,0 +1,172 @@
+    let degree: -> int = [];
+
+namespace std::utils;
+    /// Evaluates to folder(...folder(folder(initial, f(0)), f(1)) ..., f(length - 1)),
+    /// i.e. calls f(0), f(1), ..., f(length - 1) and combines the results
+    /// using the function `folder`, starting with the value `initial`.
+    ///
+    /// See `sum` for an example use.
+    let<T1, T2>
+        fold: int, (int -> T1), T2, (T2, T1 -> T2) -> T2 = |length, f, initial, folder|
+            if length <= 0 {
+                initial
+            } else {
+                folder(fold((length - 1), f, initial, folder), f((length - 1)))
+            };
+
+namespace std::array;
+    /// This is a built-in function taking an array argument and returning
+    /// the length of the array.
+    /// This symbol is not an empty array, the actual semantics are overridden.
+    let<T> len: T[] -> int = [];
+
+    /// Evaluates to the array [f(0), f(1), ..., f(length - 1)].
+    let<T> new: int, (int -> T) -> T[] = |length, f| std::utils::fold(length, f, [], |acc, e| (acc + [e]));
+
+    /// Returns a new array equal to arr except that the element at index i is x.
+    let<T> set_element: T[], int, T -> T[] = |arr, i, x| {
+        let _ = std::check::assert(i < len(arr), || "Index out of bounds");
+        map_enumerated(arr, |j, y| if i == j { x } else { y })
+    };
+
+    /// Returns a new array of length l containing the elements of arr starting at index start.
+    let<T> sub_array: T[], int, int -> T[] = |arr, start, l| std::array::new(l, |i| arr[start + i]);
+
+    /// Evaluates to the array [f(arr[0]), f(arr[1]), ..., f(arr[len(arr) - 1])].
+    let<T1, T2> map: T1[], (T1 -> T2) -> T2[] = |arr, f| new(len(arr), |i| f(arr[i]));
+
+    /// Evaluates to the array [f(0, arr[0]), f(1, arr[1]), ..., f(len(arr) - 1, arr[len(arr) - 1])].
+    let<T1, T2> map_enumerated: T1[], (int, T1 -> T2) -> T2[] = |arr, f| new(len(arr), |i| f(i, arr[i]));
+
+    /// Computes folder(...folder(folder(initial, arr[0]), arr[1]) ..., arr[len(arr) - 1])
+    let<T1, T2> fold: T1[], T2, (T2, T1 -> T2) -> T2 = |arr, initial, folder| std::utils::fold(len(arr), |i| arr[i], initial, folder);
+
+    /// Returns the sum of the array elements.
+    // TODO: Should make use of the Default or Zero trait instead of FromLiteral (then we can also
+    // use this function to flatten an array of arrays.
+    let<T: Add + FromLiteral> sum: T[] -> T = |arr| fold(arr, 0, |a, b| a + b);
+
+    /// Returns the product of the array elements.
+    let<T: Mul + FromLiteral> product: T[] -> T = |arr| fold(arr, 1, |a, b| a * b);
+
+    /// Zips two arrays
+    /// TODO: Assert that lengths are equal when expressions are supported.
+    let<T1, T2, T3> zip: T1[], T2[], (T1, T2 -> T3) -> T3[] = |array1, array2, fn| new(len(array1), |i| fn(array1[i], array2[i]));
+
+namespace Main(8);
+
+    // A ArrayTerm in an `ArrayExpression`
+    enum ArrayTerm {
+        Repeat(int[]),
+        Once(int[])
+    }
```

Comment:
```suggestion
    enum ArrayTerm {
        Repeat(ArrayTerm),
        Concatenate(ArrayTerm, ArrayTerm),
        Literal(int[]),
    }
```
?

Reply:
I can change that but is necessary? I don't think the parser lets you create `Repeat(x)` where `x` is not `int[]`